Automattic/pocket-casts-android
# Bookmarks: Adds database table and DAO

## Description

This change creates the new bookmarks table and adds a basic DAO.

I don't know if we usually hold back the migration until later in case it changes early on in the project. 

iOS has the following structure.
```
CREATE TABLE IF NOT EXISTS \(Self.tableName) (
    \(Column.uuid) varchar(40) NOT NULL,
    \(Column.episode) varchar(40) NOT NULL,
    \(Column.podcast) varchar(40),
    \(Column.time) real NOT NULL,
    \(Column.createdDate) INTEGER NOT NULL,
    \(Column.deleted) int NOT NULL DEFAULT 0,
    PRIMARY KEY (\(Column.uuid))
);
```
but the Android version is slightly different.
```
CREATE TABLE IF NOT EXISTS `bookmarks` (
    `uuid` TEXT NOT NULL,
    `podcast_uuid` TEXT NOT NULL,
    `episode_uuid` TEXT NOT NULL,
    `time` INTEGER NOT NULL, 
    `created_at` INTEGER NOT NULL,
    `deleted` INTEGER NOT NULL,
    `sync_status` INTEGER NOT NULL,
    PRIMARY KEY(`uuid`)
);
```
I have gone with the not null for everything. With user file episodes we have a `UserEpisodePodcastSubstitute.substituteUuid` that we send to the server, so I wondered if we should wrap that logic in a BookmarkManager and keep it the same. I'm open to suggestions for this.

## Testing Instructions
1. Install the previous version.
2. Install this version over the top
3. ✅ Verify this is no Room migration issues
